app/src/main/java/org/thoughtcrime/securesms/util/Util.java

```diff
@@ -0,0 +1,518 @@
+    }
+
+    StringBuilder out = new StringBuilder(value);
+    while (out.length() < length) {
+      out.append(" ");
+    }
+
+    return out.toString();
+  }
+
+  public static boolean isEmpty(EncodedStringValue[] value) {
+    return value == null || value.length == 0;
+  }
+
+  public static boolean isEmpty(ComposeText value) {
+    return value == null || value.getText() == null || TextUtils.isEmpty(value.getTextTrimmed());
+  }
+
+  public static boolean isEmpty(Collection<?> collection) {
+    return collection == null || collection.isEmpty();
+  }
+
+  public static boolean isEmpty(@Nullable String value) {
+    return value == null || value.length() == 0;
+  }
+
+  public static boolean hasItems(@Nullable Collection<?> collection) {
+    return collection != null && !collection.isEmpty();
+  }
+
+  public static <K, V> V getOrDefault(@NonNull Map<K, V> map, K key, V defaultValue) {
+    return map.containsKey(key) ? map.get(key) : defaultValue;
+  }
+
+  public static String getFirstNonEmpty(String... values) {
+    for (String value : values) {
+      if (!Util.isEmpty(value)) {
+        return value;
+      }
+    }
+    return "";
+  }
+
+  public static @NonNull String emptyIfNull(@Nullable String value) {
+    return value != null ? value : "";
+  }
+
+  public static @NonNull CharSequence emptyIfNull(@Nullable CharSequence value) {
+    return value != null ? value : "";
+  }
+
+  public static <E> List<List<E>> chunk(@NonNull List<E> list, int chunkSize) {
+    List<List<E>> chunks = new ArrayList<>(list.size() / chunkSize);
+
+    for (int i = 0; i < list.size(); i += chunkSize) {
+      List<E> chunk = list.subList(i, Math.min(list.size(), i + chunkSize));
+      chunks.add(chunk);
+    }
+
+    return chunks;
+  }
+
+  public static CharSequence getBoldedString(String value) {
+    SpannableString spanned = new SpannableString(value);
+    spanned.setSpan(new StyleSpan(Typeface.BOLD), 0,
+                    spanned.length(),
+                    Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+
+    return spanned;
+  }
+
+  public static @NonNull String toIsoString(byte[] bytes) {
+    try {
+      return new String(bytes, CharacterSets.MIMENAME_ISO_8859_1);
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError("ISO_8859_1 must be supported!");
+    }
+  }
+
+  public static byte[] toIsoBytes(String isoString) {
+    try {
+      return isoString.getBytes(CharacterSets.MIMENAME_ISO_8859_1);
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError("ISO_8859_1 must be supported!");
+    }
+  }
+
+  public static byte[] toUtf8Bytes(String utf8String) {
+    try {
+      return utf8String.getBytes(CharacterSets.MIMENAME_UTF_8);
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError("UTF_8 must be supported!");
+    }
+  }
+
+  public static void wait(Object lock, long timeout) {
+    try {
+      lock.wait(timeout);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)